app/src/main/java/com/amaze/filemanager/filesystem/SafRootHolder.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2014-2021 Arpit Khurana <arpitkh96@gmail.com>, Vishal Nehra <vishalmeham2@gmail.com>,
+ * Emmanuel Messulam<emmanuelbendavid@gmail.com>, Raymond Lai <airwave209gt at gmail.com> and Contributors.
+ *
+ * This file is part of Amaze File Manager.
+ *
+ * Amaze File Manager is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.amaze.filemanager.filesystem
+
+import android.net.Uri
+
+object SafRootHolder {
```

Comment:
Make functions here that set up the fields, then call them from where it is appropiate, as there is quite a few lines of code that need to be executed to set up the process and it should be centralized.

Reply:
Sorry but I'm a bit confused about the code for setting up the `SafRootHolder`.

Reply:
It seems this code is used: 

```java
  private ActivityResultLauncher<Intent> handleDocumentUriForRestrictedDirectories =
      registerForActivityResult(
          new ActivityResultContracts.StartActivityForResult(),
          result -> {
            if (SDK_INT >= Q) {
              getContext()
                  .getContentResolver()
                  .takePersistableUriPermission(
                      result.getData().getData(),
                      Intent.FLAG_GRANT_READ_URI_PERMISSION
                          | Intent.FLAG_GRANT_WRITE_URI_PERMISSION);
              SafRootHolder.setUriRoot(result.getData().getData());
              loadlist(result.getData().getDataString(), false, OpenMode.DOCUMENT_FILE);
            }
          });
```

Reply:
Sorry but I'm really out of ideas at this point :(

Reply:
How do you know the uses of these variables won't NPE? Is there an intuive way to check the flow?